Assets/FunnyJuice/Scripts/PlayerInteractor.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerInteractionRay : MonoBehaviour
+{
+    [SerializeField] private LayerMask interactionMask;
+
+    private GameObject previousObjectHit;
+    
+    void Update()
+    {
+        if (!Physics.Raycast(transform.position, transform.forward, out RaycastHit hit, 10f, interactionMask))
+        {
+            if(previousObjectHit != null && previousObjectHit.TryGetComponent(out FunnyJuice previousFunnyJuice))
+                previousFunnyJuice.Unhighlight();
+            
+            return;
+        }
+
+        if (hit.collider.TryGetComponent(out FunnyJuice funnyJuice))
+        {
+            previousObjectHit = funnyJuice.gameObject;
+            funnyJuice.Highlight();  
+            
+            if (Input.GetKeyDown(KeyCode.E))
```

Comment:
TODO : 
I need to replace it with a proper input control asset, to test faster I did it dirty 😅